### PR TITLE
Use certifi to tell urllib where to find CA file

### PR DESCRIPTION
**Pull request by oxve**
Also clean up some unused Docker images.

b/359451816
b/355184050
b/354270960

### CODE REVIEW

**Comment by jonastsai** [comment]
there is no certifi in requirements.in/txt file in following folders:

- docker/linux/linux-x64x11
- docker/linux/unittest/
- docker/precommit_hooks/
- docker/pytest/
- precommit_hooks/

Is it because they don't need to be prebuilt and I can add them later when I modify `cobalt/tools/download_from_gcs.py`?

Or they are not used when I modify `cobalt/tools/download_from_gcs.py`?

File: docker/linux/base/build/Dockerfile [hunk 1]
```diff
@@ -123,6 +123,10 @@ RUN cd /tmp \
     && echo 16-init-17653-g39da55e8-2 >> ${CLANG_16_TC_HOME}/cr_build_revision \
     && rm clang-llvmorg-16-init-17653-g39da55e8-2.tgz
 
+# === Install build time python dependencies
+COPY base/build/requirements.txt /opt/requirements.txt
+RUN python3 -m pip install --require-hashes --no-deps -r /opt/requirements.txt
```
Comment (kaidokert) [inline docker/linux/base/build/Dockerfile:128]: I think you are doing a no-op refactor, but given that this moved here.. shouldn't it just become an apt `python3-certifi` package above in this file ? ( note `python-certifi` probably isn't needed, Python2 should be gone )

File: docker/linux/clang-3-9/Dockerfile [hunk 2]
```diff
@@ -34,5 +34,8 @@ RUN python3.8 -m venv /python38_env
 ENV VIRTUAL_ENV /python38_env
 ENV PATH /python38_env/bin:$PATH
 
+COPY clang-3-9/requirements.txt /opt/requirements.txt
+RUN python3.8 -m pip install --require-hashes --no-deps -r /opt/requirements.txt
```
Comment (kaidokert) [inline docker/linux/clang-3-9/Dockerfile:38]: Note here we need it via pip, in bionic the latest available is from 2018

```
python3-certifi/bionic 2018.1.18-2
```

File: docker/linux/base/build/Dockerfile (hunk 1 above)
Comment (kaidokert) [reply docker/linux/base/build/Dockerfile:128]: Perhaps not, i checked the `debian11` and it has `python3-certifi/oldstable 2020.6.20-1`

Comment (oxve) [reply docker/linux/base/build/Dockerfile]: Looks like it worked.

**Review by kaidokert — APPROVED** [review]

**Pull request merged**